### PR TITLE
Created hla-la/typing and hla-la/preparegraph module

### PR DESCRIPTION
**Pull request by WackerO**
## PR checklist

Closes #XXX 

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [ ] Remove all TODO statements.
- [ ] Emit the `versions.yml` file.
- [ ] Follow the naming conventions.
- [ ] Follow the parameters requirements.
- [ ] Follow the input/output options guidelines.
- [ ] Add a resource `label`
- [ ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [ ] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

**Comment by mapo9** [comment]
Hi, 
all tests that can pass did pass.
It's not possible to run the `hlala/preparegraph` with conda, so I added some code to stop the module when trying to use conda in the same fashion as e.g. [cellranger/count](https://github.com/nf-core/modules/blob/master/modules/nf-core/cellranger/count/main.nf)

File: modules/nf-core/hlala/preparegraph/main.nf [hunk 1]
```diff
@@ -0,0 +1,33 @@
+process HLALA_PREPAREGRAPH {
+    label 'process_high'
+
+    conda (params.enable_conda ? "bioconda::hla-la=1.0.3" : null)
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/hla-la:1.0.3--hd03093a_0':
+        'quay.io/biocontainers/hla-la:1.0.3--hd03093a_0' }"
+
+    input:
+        tuple val(meta), path(graph)
```
Comment (ggabernet) [inline modules/nf-core/hlala/preparegraph/main.nf:10]:
```suggestion
    tuple val(meta), path(graph)
```

Comment (ggabernet) [reply modules/nf-core/hlala/preparegraph/main.nf:10]: here indentation looks weird

File: modules/nf-core/hlala/typing/main.nf [hunk 2]
```diff
@@ -0,0 +1,147 @@
+process HLALA_TYPING {
+    tag "$meta.id"
+    label 'process_single'
+
+    conda (params.enable_conda ? "bioconda::hla-la=1.0.3" : null)
```
Comment (ggabernet) [inline modules/nf-core/hlala/typing/main.nf:5]:
```suggestion
    conda "bioconda::hla-la=1.0.3"
```

Comment (ggabernet) [reply modules/nf-core/hlala/typing/main.nf:5]: we have a new syntax now, the params `enable_conda` is no longer needed

File: modules/nf-core/hlala/preparegraph/main.nf [hunk 3]
```diff
@@ -0,0 +1,33 @@
+process HLALA_PREPAREGRAPH {
+    label 'process_high'
+
+    conda (params.enable_conda ? "bioconda::hla-la=1.0.3" : null)
```
Comment (ggabernet) [inline modules/nf-core/hlala/preparegraph/main.nf:4]:
```suggestion
    conda "bioconda::hla-la=1.0.3"
```

Comment (ggabernet) [reply modules/nf-core/hlala/preparegraph/main.nf:4]: wasn't there the problem that it does not work with conda and we wanted to add a warning here?

File: tests/modules/nf-core/hlala/typing/main.nf [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+    SAMTOOLS_INDEX(input)
+
+    input_ch = Channel.from(input).toList()
+
+    input_ch.concat(graph, SAMTOOLS_INDEX.out.bai)
+                                            .flatten().toList()
+                                            .map { meta, cram, graph, meta_1, bai ->
+                                                    [meta, cram, bai, graph] }
+                                            .set { hla_typing_input_ch }
+    hla_typing_input_ch
```
Comment (ggabernet) [inline tests/modules/nf-core/hlala/typing/main.nf:26]:
```suggestion
```

File: tests/modules/nf-core/hlala/typing/main.nf [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+//     input_ch.concat(graph, SAMTOOLS_INDEX.out.bai)
+//                                             .flatten().toList()
+//                                             .map { meta, cram, graph, meta_1, bai ->
+//                                                     [meta, cram, bai, graph] }
+//                                             .set { hla_typing_input_ch }
+
+//     hla_typing_input_ch
+
+//     HLALA_TYPING ( hla_typing_input_ch )
+// }
```
Comment (ggabernet) [inline tests/modules/nf-core/hlala/typing/main.nf:60]:
```suggestion
```

**Review by ggabernet — COMMENTED** [review]
Just a few comments @mapo9

File: tests/modules/nf-core/hlala/typing/main.nf (hunk 5 above)
Comment (mapo9) [reply tests/modules/nf-core/hlala/typing/main.nf:59]: I added that part to document how a larger test would be done as required [here](https://nf-co.re/docs/contributing/modules#using-a-stub-test-when-required-test-data-is-too-big) and seen in the [ascat](https://github.com/nf-core/modules/blob/master/tests/modules/nf-core/ascat/main.nf#L27) module. Is that not required anymore?

File: tests/modules/nf-core/hlala/typing/main.nf [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+
+    HLALA_TYPING ( hla_typing_input_ch )
+}
+
+ // extended tests running with actual test data (also see https://github.com/DiltheyLab/HLA-LA)
+    
+// Download files
+// 
+// wget http://www.well.ox.ac.uk/downloads/PRG_MHC_GRCh38_withIMGT.tar.gz
+// wget https://www.dropbox.com/s/xr99u3vqaimk4vo/NA12878.mini.cra
```
Comment (ggabernet) [inline tests/modules/nf-core/hlala/typing/main.nf:35]:
```suggestion
// wget https://www.dropbox.com/s/xr99u3vqaimk4vo/NA12878.mini.cram
```

**Review by ggabernet — APPROVED** [review]
Looks good to me now!

**Pull request merged**